README.md

```diff
@@ -1,27 +1,201 @@
 
-```bash
-golem-sdk --help
+
+### Creating a Golem Manifest
+
+To create a new Golem Manifest with `golem-sdk` CLI, run:
+
+```shell
+golem-sdk manifest create <image>
 ```
+
+If you have a `package.json` file in your project, it will be used to fill in the `name`, `version` and `description` fields of the manifest. Otherwise you wiil need to provide them manually.
+
+Provided `image` argument should identify the GVMI image that will be used by your application.
+
+
+#### Image
+
+The manifest needs to contain the image URL pointing to GVMI download location and it's hash to validate its integrity.
+In order to facilitate the process of creating a manifest, `golem-sdk` accepts multiple forms of image argument, where some of them will automatically resolve the URL and/or hash.
+Please consult the table bellow for more details:
+
+| Argument                           | `--image-hash`               | Example                                                                                       | Notes                                                                                             |
+|------------------------------------|------------------------------|-----------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------|
+| Image tag                          | automatically resolved       | `golem/node:latest`                                                                           | Image hash will be fetched from [https://registry.golem.network]. This is the recommended method. |
+| Image hash                         | resolved from image argument | `3d6c48bb4c192708168d53cee4f36876b263b7745c3a3c239c6749cd`                                    | Image URL will point to [https://registry.golem.network]                                          |
+| URL to registry.golem.network      | automatically resolved       | `https://registry.golem.network/v1/image/download?tag=golem-examples/blender:2.80&https=true` |                                                                                                   |
+| URL to arbitrary download location | hash is needed               | `https://example.com/my-image`                                                                |                                                                                                   |
+
+If hash is not provided or resolved, you will get a warning that the manifest will not be usable until you provide it manually.
+
+
+### Adding outbound URLs
+
+In order to be able to access the internet from Golem network, your application needs to declare the outbound URLs it will be using inside its manifest.
+
+There is a default set of URLs that providers may allow your application to use (TODO: link to default URL whitelist).
```

Comment:
Whitelisted domains/ips can be found here

https://github.com/golemfactory/ya-installer-resources/tree/main/whitelist